src/relax/transform/static_plan_block_memory.cc

```diff
@@ -336,6 +336,13 @@ class StorageAllocatorInit : public StorageAllocatorBaseVisitor {
     // appear in the **function signature**.
     Map<String, IntImm> var_upper_bound_attr =
         func->GetAttr<Map<String, IntImm>>("tir_var_upper_bound").value_or(Map<String, IntImm>());
+    // `GetAttr` does not perform type check.
```

Comment:
“IsInstance”-style type checking, if can be determined in compile time, will be folded to a constant without actually doing the checking in runtime. Therefore, to ensure runtime type checking happens, we will need to use a step-phase downcasting, where the first phase casts an attribute to Map<ObjectRef, ObjectRef>, and then check the types of each item